ts/features/idpay/wallet/utils/hooks.ts

```diff
@@ -0,0 +1,47 @@
+import { useFocusEffect } from "@react-navigation/native";
+import * as React from "react";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import { idPayInitiativesFromInstrumentGet } from "../store/actions";
+import {
+  idPayAreInitiativesFromInstrumentErrorSelector,
+  idPayEnabledInitiativesFromInstrumentSelector
+} from "../store/reducers";
+
+export const useIDPayInitiativesFromInstrument = (idWallet: string) => {
```

Comment:
Wouldn't be a simpler solution organizing initiatives by walletID instead of set/reset data each time? We could also have some caching in that wat. What do you think?